serde/src/java/org/apache/hadoop/hive/serde2/objectinspector/primitive/WritableConstantHiveDecimalObjectInspector.java

```diff
@@ -62,13 +62,4 @@ public int precision() {
     }
```

Comment:
Why does precision need to be overridden here (while scale does not)? Could we possibly remove this too?

Reply:
We can remove it and it can also be removed from `WritableConstant[Byte|Int|Long|Short]ObjectInspector`.
But it doesn't have such an effect like `scale()`. `scale()` plays a role in converting to String but `precision()` is used only in functions `GenericUDFBaseNumeric`, `GenericUDAFAverage`, `GenericUDAFSumHiveDecimal` to determine the result type.

Reply:
Seems that removing precision override has unwanted side effects. I reverted that part.